staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -1049,6 +1063,33 @@ func (cfgCtlr *configController) maybeReapReadLocked(plName string, plState *pri
 	cfgCtlr.configQueue.Add(0)
 }
 
+// PriorityLevelMaxSeats returns a functions invoked by the work estimator
+// to dynamically determine max seats for a request in a given priority level.
+// See priorityLevelMaxSeatsLocked for more details.
+func (cfgCtlr *configController) PriorityLevelMaxSeats() fcrequest.MaxSeatsFunc {
+	return func(plName string) uint64 {
+		// This function should only be invoked when work estimator runs in startRequest,
+		// where cgfCtlr's lock is already acquired.
```

Comment:
This should be function-level comment.

Also - I'm still wondering if this is not too fragile...